kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/NamedRange.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config;
+
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Represents the set of integers between two integer endpoints. So the Range
+ * with startInclusive 1 and endExclusive 3 contains the integers 1 and 2.
+ * A Range must be non-empty, endExclusive must be greater than startInclusive
+ */
+public record NamedRange(@NonNull String name,
+                         @JsonInclude(JsonInclude.Include.ALWAYS) int startInclusive,
+                         @JsonInclude(JsonInclude.Include.ALWAYS) int endExclusive) {
```

Comment:
Since we only using this new type for the new `listeners`/`networkPresentations` tree... I think most people would assume that plain `start` and `end` would be inclusive. It's also weird to have to write down a node id for a node that you know doesn't exist, just because the end is exclusive.
So I would advocate for just `start` and `end`, sure it's a little less explicit, but it's shorter and more readable over all.